Browserstack: Build not created, Error: connect ECONNREFUSED

<!--
## Have you read the "Debugging SDKs" doc yet?

There are common setup gotchas that happen with Percy's SDKs, it would be worth reading
the debugging document, which might already answer your question:
https://docs.percy.io/docs/debugging-sdks

## Reach out to Percy support instead?

If you’re having issues that _aren’t SDK bugs_, it would be best for you to
reach out to support instead: support@percy.io or
https://www.browserstack.com/contact#technical-support
-->

## The problem

Recently, without having made any changes to the environment, our GitHub Actions visual regression testing using Browserstack stopped working. This could be related to https://github.com/percy/cli/issues/1459.

## Environment

This happens in GitHub Actions.

- Node version: built-in actions version
- `@percy/cli` version: latest release 
- OS version: ubuntu-latest

## Debug logs
```
Run npx percy snapshot _site/ -v
  npx percy snapshot _site/ -v
  shell: /usr/bin/bash -e {0}
  env:
    PERCY_TOKEN: ***
[percy:config] Config file not found (0ms)
[percy:core:install] Downloading Chromium 9295[1](https://github.com/hydephp/develop/actions/runs/7150918096/job/19474699781?pr=1487#step:6:1)1...
[percy:core:install] Successfully downloaded Chromium 9[2](https://github.com/hydephp/develop/actions/runs/7150918096/job/19474699781?pr=1487#step:6:2)9511 ([3](https://github.com/hydephp/develop/actions/runs/7150918096/job/19474699781?pr=1487#step:6:3)695ms)
[percy:core:browser] Launching browser (29ms)
[percy:core:browser] Browser connected [1993]: HeadlessChrome/96.0.[4](https://github.com/hydephp/develop/actions/runs/7150918096/job/19474699781?pr=1487#step:6:4)664.0 (382ms)
[percy:core] Percy has started! (0ms)
[percy:core:queue] Clearing discovery queue, queued state: 0, pending state: 0 (272ms)
[percy:core:queue] Clearing snapshot queue, queued state: 0, pending state: 0 (0ms)
[percy:core] Stopping percy... (0ms)
[percy:core:queue] Clearing discovery queue, queued state: 0, pending state: 0 (0ms)
[percy:core:browser] Closing browser (1ms)
[percy:core:browser] Browser closed (1[5](https://github.com/hydephp/develop/actions/runs/7150918096/job/19474699781?pr=1487#step:6:5)ms)
[percy:core:queue] Clearing snapshot queue, queued state: 0, pending state: 0 (0ms)
[percy:core] Build not created (1ms)
[percy:cli] Error: connect ECONNREFUSED ::1:40751
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1555:1[6](https://github.com/hydephp/develop/actions/runs/7150918096/job/19474699781?pr=1487#step:6:7)) (0ms)
Error: Process completed with exit code 1.
```

## Code to reproduce issue

```yaml
- name: Install Percy
  run: npm install --save-dev @percy/cli

- name: Upload build to Percy
  run: npx percy snapshot _site/ -v
  env:
    PERCY_TOKEN: ${{ secrets.PERCY_TOKEN }}
```

Please see test run https://github.com/hydephp/develop/actions/runs/7150918096/job/19474699781?pr=1487

--- a/packages/core/src/network.js
+++ b/packages/core/src/network.js
@@ -171,6 +171,7 @@ export class Network {
     this.#pending.delete(requestId);
 
     // guard against redirects with the same requestId
+    // eslint-disable-next-line babel/no-unused-expressions
     pending?.request.url === event.request.url &&
     pending.request.method === event.request.method &&
     await this._handleRequest(session, { ...pending, resourceType, interceptId });

--- a/packages/core/src/server.js
+++ b/packages/core/src/server.js
@@ -114,9 +114,9 @@ export class Server extends http.Server {
     });
   }
 
-  // return host bind address - defaults to 0.0.0.0
+  // return host bind address - defaults to "::"
   get host() {
-    return process.env.PERCY_SERVER_HOST || '0.0.0.0';
+    return process.env.PERCY_SERVER_HOST || '::';
   }
 
   // return the listening port or any default port
@@ -127,11 +127,11 @@ export class Server extends http.Server {
   // return a string representation of the server address
   address() {
     let port = this.port;
-    // we need to specifically map 0.0.0.0 to localhost on windows as even though we
-    // can listen to all interfaces using 0.0.0.0 we cant make a request on 0.0.0.0 as
+    // we need to specifically map "::" to localhost on windows as even though we
+    // can listen to all interfaces using "::" we cant make a request on "::" as
     // its an invalid ip address as per spec, but unix systems allow request to it and
     // falls back to localhost
-    let host = `http://${this.host === '0.0.0.0' ? 'localhost' : this.host}`;
+    let host = `http://${this.host === '::' ? 'localhost' : this.host}`;
     return port ? `${host}:${port}` : host;
   }
 

--- a/packages/core/test/unit/server.test.js
+++ b/packages/core/test/unit/server.test.js
@@ -22,7 +22,7 @@ describe('Unit / Server', () => {
 
   describe('#host', () => {
     it('returns the host', async () => {
-      expect(server.host).toEqual('0.0.0.0');
+      expect(server.host).toEqual('::');
     });
 
     describe('with PERCY_SERVER_HOST set', () => {

